cpp/src/decisiontree/levelalgo/levelfunc_classifier.cuh

```diff
@@ -109,10 +109,11 @@ void grow_deep_tree_classification(
   }
   std::vector<unsigned int> feature_selector(h_colids, h_colids + Ncols);
 
-  int scatter_algo_depth = std::min(tempmem->swap_depth, tree_params.max_depth);
+  int scatter_algo_depth =
+    std::min(tempmem->swap_depth, tree_params.max_depth + 1);
   for (int depth = 0; (depth < scatter_algo_depth) && (n_nodes_nextitr != 0);
        depth++) {
-    depth_cnt = depth + 1;
+    depth_cnt = depth;
```

Comment:
`depth_cnt` is used in `print_summary()` and `print_detailed()`, and this change is necessary so that these two print functions produce the correct depth.

Reply:
gotcha, makes sense now